test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -333,6 +334,7 @@ type ApplyCustomClusterTemplateAndWaitInput struct {
 	ClusterName                  string
 	Namespace                    string
 	CNIManifestPath              string
+	Flavor                       string
```

Comment:
Let's add a godoc comment here that we are just adding the flavor field because this struct is also used by the Waiter interface and that ApplyCustomClusterTemplateAndWait is not using it